zebrad/tests/common/get_block_template_rpcs.rs

```diff
@@ -0,0 +1,164 @@
+        }
+
+        let _ = child.kill(true);
+        let copied_state_path = child.dir.take().unwrap();
+
+        let (_read_write_state_service, mut state, _latest_chain_tip, _chain_tip_change) =
+            start_state_service_with_cache_dir(network, copied_state_path.as_ref()).await?;
+        let request = zebra_state::ReadRequest::Block(HashOrHeight::Height(future_block_height));
+
+        let response = state
+            .ready()
+            .and_then(|ready_service| ready_service.call(request))
+            .map_err(|error| eyre!(error))
+            .await?;
+
+        let block_hex_data = match response {
+            zebra_state::ReadResponse::Block(Some(block)) => {
+                hex::encode(block.zcash_serialize_to_vec()?)
+            }
+            zebra_state::ReadResponse::Block(None) => {
+                tracing::info!(
+                "Reached the end of the finalized chain, state is missing block at {future_block_height:?}",
+            );
+                return Ok(None);
+            }
+            _ => unreachable!("Incorrect response from state service: {response:?}"),
+        };
+
+        Ok(Some(block_hex_data))
+    }
+
+    #[allow(clippy::print_stderr)]
+    pub(crate) async fn run() -> Result<(), color_eyre::Report> {
+        let _init_guard = zebra_test::init();
+
+        let mut config = random_known_rpc_port_config(true)?;
+        let network = config.network.network;
+        let rpc_address = config.rpc.listen_addr.unwrap();
+
+        config.state.cache_dir = match std::env::var_os(ZEBRA_CACHED_STATE_DIR) {
+            Some(path) => path.into(),
+            None => {
+                eprintln!(
+                    "skipped submitblock test, \
+                     set the {ZEBRA_CACHED_STATE_DIR:?} environment variable to run the test",
+                );
+
+                return Ok(());
+            }
+        };
+
+        // TODO: As part of or as a pre-cursor to issue #5015,
+        // - Use only original cached state,
+        // - sync until the tip
+        // - get first 3 blocks in non-finalized state via getblock rpc calls
+        // - restart zebra without peers
+        // - submit block(s) above the finalized tip height
+        let block_hex_data = get_future_block_hex_data(network, &config.state.cache_dir)
+            .await?
+            .expect("spawned zebrad in get_future_block_hex_data should live until it gets the next block");
+
+        // Runs the rest of this test without an internet connection
+        config.network.initial_mainnet_peers = IndexSet::new();
+        config.network.initial_testnet_peers = IndexSet::new();
+        config.mempool.debug_enable_at_height = Some(0);
+
+        // We're using the cached state
+        config.state.ephemeral = false;
+
+        let mut child = testdir()?
+            .with_exact_config(&config)?
+            .spawn_child(args!["start"])?
+            .bypass_test_capture(true);
```

Comment:
This code seems to be a duplicate of `spawn_zebrad_for_rpc()`, but it is missing a timeout.

This will make the test run forever if Zebra hangs, or the test conditions are wrong.

Reply:
I'll update this in the PR for #5015 when adding it to CI